backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java

```diff
@@ -845,18 +844,18 @@ IGitMacheteRepositorySnapshot discoverLayoutAndCreateSnapshot(int mostRecentlyCh
             .sortBy(branchName -> lastCheckoutTimestampByBranchName.getOrElse(branchName, Instant.MIN))
             .reverse()
             .splitAt(mostRecentlyCheckedOutBranchesCount);
-        freshNonFixedRootBranchNames = freshAndStaleNonFixedRootBranchNames._1;
+        freshNonFixedRootBranchNames = freshAndStaleNonFixedRootBranchNames._1.sorted();
 
         LOG.debug(() -> "Skipping stale branches from the discovered layout: "
             + freshAndStaleNonFixedRootBranchNames._2.mkString(", "));
       }
 
-      // Let's use SortedMaps to ensure a deterministic result.
-      SortedMap<String, MyBranchLayoutEntry> entryByFixedRootBranchNames = fixedRootBranchNames
-          .toSortedMap(name -> Tuple.of(name, new MyBranchLayoutEntry(name)));
-      SortedMap<String, MyBranchLayoutEntry> entryByFreshNonFixedRootBranch = freshNonFixedRootBranchNames
-          .toSortedMap(name -> Tuple.of(name, new MyBranchLayoutEntry(name)));
-      SortedMap<String, MyBranchLayoutEntry> entryByIncludedBranchName = entryByFixedRootBranchNames
+      // Let's use linked maps to ensure a deterministic result.
+      Map<String, MyBranchLayoutEntry> entryByFixedRootBranchNames = fixedRootBranchNames
+          .toLinkedMap(name -> Tuple.of(name, new MyBranchLayoutEntry(name)));
+      Map<String, MyBranchLayoutEntry> entryByFreshNonFixedRootBranch = freshNonFixedRootBranchNames
+          .toLinkedMap(name -> Tuple.of(name, new MyBranchLayoutEntry(name)));
+      Map<String, MyBranchLayoutEntry> entryByIncludedBranchName = entryByFixedRootBranchNames
```

Comment:
Aaand here's the part where an existing inconsistency with CLI has been fixed... it's pretty subtle so I haven't `discovered` (no pun intended) until I changed the conditions for `develop` being a root